identity-credential/src/credential/credential.rs

```diff
@@ -79,6 +82,9 @@ pub struct Credential<T = Object> {
   /// Miscellaneous properties.
   #[serde(flatten)]
   pub properties: T,
+  /// Proof(s) used to verify a `Credential`
+  #[serde(skip_serializing_if = "Option::is_none")]
+  pub proof: Option<Signature>,
```

Comment:
Does this actually get populated when deserialized or is the `proof` field "merged" into `properties` because of `#[serde(flatten)]`?

Reply:
I thought the same when diagnosing why the tests that deserialize Credentials and Presentations from Json failed. However, those run through, i.e. here: 
https://github.com/iotaledger/identity.rs/blob/58608f86f2c9c494784ce2f856bcb343fa9880e2/identity-credential/src/presentation/presentation.rs#L173

When I noticed there seems to be no difference, in terms of flatten, I put it in the last position since this would have been the original layout after the inner type was flattened in the Verifiable* version which also matches the specs.

The docs on flatten also mention that you can use flatten multiple times, if our initial suspicion was correct, this would never be possible?

Reply:
looking at the generated code it doesn't seem to be an issue:

```rust
let __field14 = match __field14 {
    _serde::__private::Some(__field14) => __field14,
    _serde::__private::None => {
        match _serde::__private::de::missing_field("proof") {
            _serde::__private::Ok(__val) => __val,
            _serde::__private::Err(__err) => {
                return _serde::__private::Err(__err);
            }
        }
    }
};
let __field13: T = match _serde::de::Deserialize::deserialize(
    _serde::__private::de::FlatMapDeserializer(
        &mut __collect,
        _serde::__private::PhantomData,
    ),
) {
    _serde::__private::Ok(__val) => __val,
    _serde::__private::Err(__err) => {
        return _serde::__private::Err(__err);
    }
};
```

but we should definitely have more tests 👍